src/Preg.php

```diff
@@ -35,6 +35,8 @@ final class Preg
      */
     public static function match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
     {
+        $pattern = self::verify($pattern);
```

Comment:
I still don't understand why is the assignment here - simply `self::verify($pattern);` would work the same.

Reply:
@kubawerlos What would help you to gain more understanding? A method does not represent a process result as much as a function that returns (spoiler: a method has no result). What is there to add? Should we go more abstract into math/logic here? What would help you?

Reply:
it was nothing about complexity, math or psalm.

I believe Kuba suggested this change:
```diff
-private static function verify($pattern): string
+private static function verify($pattern): void
```
as the only outcome of validation method is telling if input is valid or not, which `verify` method does by throwing the exception or not. no need to return anything

Reply:
This is (only) true if there is a try-catch block which re-throws. I normally spare that by making use of the return value for the happy-path. I don't say it's perfect, but much more expressive if try-catch is left out as it was suggested.

And don't spare psalm, it's in PR already.

Reply:
> much more expressive

You meant less expressive. If there is a return value person who sees it expect that this is different from the argument and would go into the function to see - otherwise can assume it's only to verify and does not change the variable.

Reply:
I see this more as a gate the data passes through. If the sub-process starts to misbehave this is more stable and stringent that is visible upfront which makes it more expressive for me, however I do see that there might be different views on that and also about it's expressiveness.

I care more about the data processing here than counting the number of code expressions.

Reply:
@kubawerlos: That method is gone. Me assumes this addresses your comment.